apps/namadillo/src/App/Ibc/IbcTransfer.tsx

```diff
@@ -99,10 +94,21 @@ export const IbcTransfer: React.FC = () => {
     );
   }, [defaultAccounts, shielded]);
 
-  const { data: ibcChannels } = useAtomValue(
-    ibcChannelsFamily(registry?.chain.chain_name)
-  );
+  const requiresIbcChannels =
+    !ibcChannels?.cosmosChannelId ||
+    (shielded && !ibcChannels?.namadaChannelId);
+
+  useEffect(() => setSelectedAssetAddress(undefined), [registry]);
 
+  // Update transaction if its hash is known and it exists in stored transactions
+  useEffect(() => {
+    if (transaction?.hash) {
+      const tx = findByHash(transaction.hash);
+      tx && setTransaction(tx);
```

Comment:
Thinking loud, how about if we create a new route/page that handle this tx progress? Then, this useEffect executes an route redirect

Sometimes I had a not so good UX when I'm on the progress page, then I click on menu Ibc button to execute a second Ibc and it doesn't navigates. Same if I click on the back button of the browser

Maybe we should just go directly to the `History` single page, like `/transaction/{hash}`

Reply:
We already have this page. However, if the transaction fails for some reason, we're returning to the previous screen, so the user can change the params. I think it would be too complicate to handle this case if we're redirecting the users to a new page directly. Any ideas?